### PR TITLE
added json request to nodes and network

### PR DESCRIPTION
**Pull request by jfeiglin**

### CODE REVIEW

File: network/network.py [hunk 1]
```diff
@@ -48,7 +48,70 @@ def _generate_nodes(self):
     def refresh_network(self):
         self.nodes = self._generate_nodes()
 
+    def get_node_by_mac(self, mac_addr):
```
Comment (benjaminbaker-dev) [inline network/network.py:51]: i dont need this function - I need:
start_mitm(mac_addr) 
stop_mitm(mac_addr)
get_tags(mac_addr)

**Review by benjaminbaker-dev — CHANGES_REQUESTED** [review]
looks good - just need more functions

File: network/network_node.py [hunk 2]
```diff
@@ -63,6 +70,42 @@ def to_json(self):
...
+        for filter_name, filter_function in type(self).SUPPORTED_FILTERS.items():
+            response[filter_name] = list(inspect.signature(filter_function).parameters)
+        return response
+
+    def json_add_rule(self, node_json_request):
+        """
+        Add rule requested by the submitted node_json_request
+        :param node_json_request: json of the form:
+            {
+            filter_name: [filter_param_value_1, filter_param_value_2, ...],
```
Comment (benjaminbaker-dev) [inline network/network_node.py]: better if this is filter_name: {param_name, param_value} no ?

**Review by benjaminbaker-dev — CHANGES_REQUESTED** [review]
d

File: network/network.py [hunk 3]
```diff
@@ -48,7 +48,70 @@ def _generate_nodes(self):
...
+        return json.dumps(response_json)
+
+    def json_node_request_add_rule(self, query_json):
+        """
+        Request a specific node to add a rule
+        :param query_json: a json of the form:
+            {
+            "node_id":<mac_of_node>,
+            "request":{
+                "filter_name":["filter_arg_1", "filter_arg_2", ...],
```
Comment (benjaminbaker-dev) [inline network/network.py]: should be filter_name: {filter_arg: value}

**Review by benjaminbaker-dev — CHANGES_REQUESTED** [review]

File: network/network.py (hunk 1 above)
Comment (jfeiglin) [reply network/network.py:51]: This isnt for you, asshole

File: network/network.py (hunk 3 above)
Comment (jfeiglin) [reply network/network.py]: thats completely unnecessary, you can just pass them as args instead of kwargs but whatever

**Review by benjaminbaker-dev — APPROVED** [review]

**Pull request merged**